pkg/loghttp/push/otlp.go

```diff
@@ -70,6 +75,18 @@ func extractLogs(r *http.Request, maxRecvMsgSize int, pushStats *Stats) (plog.Lo
 		defer func(reader *gzip.Reader) {
 			_ = reader.Close()
 		}(r)
+	case zstdContentEncoding:
+		var err error
+		body, err = zstd.NewReader(body)
+		if err != nil {
+			return plog.NewLogs(), err
```

Comment:
I am surprised that in case of an error we still return `plog.NewLogs()` as well. I this later used / is this a contract? Maybe worth adding to the function comment then, because usually, on error, the other return arguments are not supposed to be used.

Reply:
Thanks for clarifying!